src/main/java/maksym/fedorenko/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -99,4 +101,21 @@ private void checkIfBookExistsById(Long id) {
             throw new EntityNotFoundException("Book with id=%d doesn't exist".formatted(id));
         }
     }
+
+    private Predicate getFilterPredicate(BookSearchParametersDto searchDto, QBook book) {
```

Comment:
this method seems to be irrelevant to this service

consider creating a separate component like `BookSearchPredicateConstructor` with a method `construct`